notebooks/use-cases/dataset-preparation-for-RAG.ipynb

```diff
@@ -0,0 +1,892 @@
+   "source": [
+    "## ⚠️ Important Notes\n",
+    "\n",
+    "**Exception Handling**: This notebook demonstrates the core workflow with minimal error handling for clarity. When using your own data or deploying to production:\n",
+    "\n",
+    "- Add try-except blocks around file I/O operations\n",
+    "- Handle network errors for URL-based document loading\n",
+    "- Validate document formats and sizes before processing\n",
+    "- Implement timeouts for long-running operations\n",
+    "- Add proper logging for debugging and monitoring\n",
+    "- Handle cases where documents fail to convert or chunk\n",
+    "- Validate embedding generation and database insertion success\n",
+    "\n",
+    "Example of adding exception handling:\n",
+    "```python\n",
+    "try:\n",
+    "    result = converter.convert(file_path)\n",
+    "    document = result.document\n",
+    "except Exception as e:\n",
+    "    print(f\"❌ Failed to convert {file_path}: {str(e)}\")\n",
+    "    continue  # Skip to next document\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6c51bd3a",
+   "metadata": {},
+   "source": [
+    "## 🔧 Configuration\n",
+    "\n",
+    "### Set Input Documents\n",
+    "\n",
+    "Define the documents to process. You can mix URLs and local file paths.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "494f1a71",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Documents to process - replace with your own\n",
+    "input_files = [\n",
+    "    \"https://github.com/docling-project/docling/raw/v2.43.0/tests/data/pdf/2203.01017v2.pdf\",\n",
+    "    \"https://drive.google.com/file/d/1FO4ymkqK6cT5LM2x87xZvahwp6rmLUPd/view?usp=drive_link\",\n",
+    "    \"https://drive.google.com/file/d/1rkQ7vDLTekQEIaKd6EyQM5zCaSXGqmNG/view?usp=drive_link\",\n",
+    "]\n",
+    "\n",
+    "# Output directory for intermediate files\n",
+    "from pathlib import Path\n",
+    "output_dir = Path(\"data/rag-dataset-preparation/output\")\n",
+    "output_dir.mkdir(parents=True, exist_ok=True)\n",
+    "\n",
+    "print(f\"✓ Will process {len(input_files)} documents\")\n",
+    "print(f\"✓ Output directory: {output_dir.resolve()}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ac2b1799",
+   "metadata": {},
+   "source": [
+    "## 📄 Step 1: Document Conversion with Docling\n",
+    "\n",
+    "### RAG-Optimized Conversion Configuration\n",
+    "\n",
+    "For RAG systems, we configure Docling with specific settings:\n",
+    "\n",
+    "- **Table Structure Detection**: Essential for preserving tabular data relationships\n",
+    "- **OCR (Optional)**: Enables text extraction from scanned documents\n",
+    "  - **Note**: OCR is disabled by default since most PDFs are digital\n",
+    "  - Set `USE_OCR = True` below if you have scanned documents\n",
+    "  - Requires Tesseract: `brew install tesseract tesseract-lang` (macOS) or `apt-get install tesseract-ocr` (Linux)\n",
```

Comment:
Take a look at the instructions I added about Tesseract in the standard conversion notebook. TL;DR link to the Docling's installation instructions and assume you're running this notebook from a RHEL-based system (we actually will be, once we have a Workbench image in place).